lua/src/factory.cc

```diff
@@ -128,7 +128,7 @@ io::endpoint* factory::new_endpoint(
              << "' configuration field because its value is empty";
     std::string t((type.isNull())
                   ? "string" : type.toElement().text().toStdString());
-    if (t == "string")
+    if (t == "string" || t == "password")
```

Comment:
IMHO "password" is not a type and should be only be available from the UI (if needed).

Reply:
On UI side, you know the configuration generator is very generic and it does not look pretty to make a hook for passwords.
And to finish, one day I hope passwords will be encrypted, and that day it will have to be a specific type because it'll contain the password and surely other data to decode it.

To sum up, I think that if I keep the check only on string, passwords won't be used for a long time.

But maybe, another solution could be :
`if (t == "number") {
  ...
}
else if (t == "string") {
 ....
}
else if (t == "blah blah") {
...
} ...
else {
Here we set a string.
}`

And string would be a default solution.

Reply:
After discussion with Centreon Web dev team, it seems indeed that having a "password" type in the UI with a different type for Broker is difficult. That's enough for me.